File: ICDB/src/main/java/cipher/RNG.java
package cipher;

import java.security.SecureRandom;

/**
 * <p>
 *      A Secure Random Number Generator (RNG)
 * </p>
 * Created on 7/8/2016
 *
 * @author Dan Kondratyuk
 */
public class RNG {

    private static final SecureRandom random = new SecureRandom();

    public static int randomInt() {
        return random.nextInt(Integer.MAX_VALUE);
    }

}

File: ICDB/src/main/java/verify/serial/Rng.kt
package verify.serial

import java.security.SecureRandom

/**
 * A random number generator (RNG)
 *
 * Created on 8/5/2016
 * @author Dan Kondratyuk
 */
object Rng {

    private val random = SecureRandom()

    fun next(): Long = random.nextInt(Integer.MAX_VALUE).toLong()
}